crates/bridge-tx-builder/src/errors.rs

```diff
@@ -11,19 +11,19 @@ use thiserror::Error;
 #[derive(Debug, Error)]
 pub enum BridgeTxBuilderError {
     /// Error building the Deposit Transaction.
-    #[error("could not build deposit transaction")]
+    #[error("could not build deposit transaction caused by: {0}")]
     DepositTransaction(#[from] DepositTransactionError),
 
     /// Error building the Deposit Transaction.
-    #[error("could not build cooperative withdrawal transaction")]
+    #[error("could not build cooperative withdrawal transaction caused by: {0}")]
     CooperativeWithdrawalTransaction(#[from] CooperativeWithdrawalError),
 
     /// Error due to there being no script provided to create a taproot address.
     #[error("noscript taproot address for only script path spend is not possible")]
     EmptyTapscript,
 
     /// Error while building the taproot address.
-    #[error("could not build taproot address")]
+    #[error("could not build taproot address caused by: {0}")]
```

Comment:
Putting the `caused by` in these error strings is somewhat verbose, just adding the `: {0}` at the end is fine.